src/Groups/action.jl

```diff
@@ -0,0 +1,237 @@
+#############################################################################
+##
+##  common actions of group elements
+##
+##  The idea is to delegate the action of `GAPGroupElem` objects
+##  on `GAP.GapObj` objects to the corresponding GAP action,
+##  and to implement the action on native Julia objects case by case.
+
+export on_points, on_right, on_tuples, on_sets, permuted
+
+
+"""
+    on_points(pnt::GAP.GapObj, x::GAPGroupElem)
+    on_points(pnt::GAPGroupElem, x::GAPGroupElem)
+    on_points(pnt::Int, x::PermGroupElem)
+
+Return the image of `pnt` under `x`.
+If `pnt` is a `GAP.GapObj` then the action is given by `GAP.Globals.OnPoints`.
+If `pnt` is a `GAPGroupElem` then the action is given by conjugation,
+that is, the result is `x`^-1 * `pnt` * `x`.
+If `pnt` is a positive integer and `x` is a permutation then the action
+is given by the natural action that maps `pnt` to `x(pnt)`.
+
+# Examples
+```jldoctest
+julia> g = symmetric_group(3);  g[1]
+(1,2,3)
+
+julia> g[2]
+(1,2)
+
+julia> on_points(g[2], g[1])
+(2,3)
+
+julia> on_points(g[2].X, g[1])
+GAP: (2,3)
+
+julia> on_points(1, g[1])
+2
+
+```
+"""
+on_points(pnt::GAP.GapObj, x::GAPGroupElem) = GAP.Globals.OnPoints(pnt, x.X)
+
+on_points(pnt::GAPGroupElem, x::GAPGroupElem) = inv(x) * pnt * x
```

Comment:
That's inefficient, e.g. for permutations, conjugation can be computed much more efficiently than that. So I think it really should be this:
```suggestion
on_points(pnt::GAPGroupElem, x::GAPGroupElem) = pnt ^ x
```